### PR TITLE
Porting loftctl space commands

### PR DESCRIPTION
**Pull request by johannesfrey**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves ENG-3730

**Please provide a short message that should be published in the vcluster release notes**
Port space related commands from loftctl to vclusterctl.

**What else do we need to know?** 
@ThomasK33 please ping me before merge so I can squash.

### CODE REVIEW

File: cmd/vclusterctl/cmd/platform/create/space.go [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	clusterv1 "github.com/loft-sh/agentapi/v4/pkg/apis/loft/cluster/v1"
+	managementv1 "github.com/loft-sh/api/v4/pkg/apis/management/v1"
+	storagev1 "github.com/loft-sh/api/v4/pkg/apis/storage/v1"
+	"github.com/loft-sh/api/v4/pkg/product"
+	"github.com/loft-sh/loftctl/v4/cmd/loftctl/cmd/create"
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/platform/create/space.go:13]: Will be removed in a follow-up, as [`create_platform.go`](https://github.com/loft-sh/vcluster/blob/main/pkg/cli/create_platform.go#L16) currently also depends on it.

File: cmd/vclusterctl/cmd/platform/create/create.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// NewCreateCmd creates a new cobra command
+func NewCreateCmd(globalFlags *flags.GlobalFlags, defaults *pdefaults.Defaults) *cobra.Command {
+	description := product.ReplaceWithHeader("create", "")
+	if upgrade.IsPlugin == "true" {
+		description = `
+####################################################################
+##################### vcluster platform create #####################
+####################################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/create/create.go:20]: This isn't needed here

File: cmd/vclusterctl/cmd/platform/create/space.go [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+########################################################
+Creates a new space for the given project, if
+it does not yet exist.
+Example:
+devspace create space myspace
+devspace create space myspace --project myproject
+devspace create space myspace --project myproject --team myteam
+########################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/create/space.go:99]: Also not needed.

File: cmd/vclusterctl/cmd/platform/delete/delete.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+// NewDeleteCmd creates a new cobra command
+func NewDeleteCmd(globalFlags *flags.GlobalFlags, defaults *pdefaults.Defaults) *cobra.Command {
+	description := product.ReplaceWithHeader("delete", "")
+	if upgrade.IsPlugin == "true" {
+		description = `
+####################################################################
+##################### vcluster platform delete #####################
+####################################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/delete/delete.go:20]: Not needed

File: cmd/vclusterctl/cmd/platform/delete/space.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+############### devspace delete space #################
+#######################################################
+Deletes a space from a cluster
+
+Example:
+devspace delete space myspace
+devspace delete space myspace --project myproject
+#######################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/delete/space.go:61]: Not needed

File: cmd/vclusterctl/cmd/platform/import/space.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+#######################################################
+################ devspace import space ################
+#######################################################
+Imports a space into a vCluster platform project.
+Example:
+devspace import space my-space --cluster connected-cluster \
+  --project my-project --importname my-space
+#######################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/import/space.go:55]: Not needed

File: cmd/vclusterctl/cmd/platform/list/spaces.go [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+		description = `
+########################################################
+################ devspace list spaces ##################
+########################################################
+List the vCluster platform spaces you have access to
+Example:
+devspace list spaces
+########################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/list/spaces.go:49]: Not needed

File: cmd/vclusterctl/cmd/platform/share/space.go [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+########################################################
+Shares a space with another vCluster platform user or team. The user
+or team need to have access to the cluster.
+Example:
+devspace share space myspace
+devspace share space myspace --project myproject
+devspace share space myspace --project myproject --user admin
+########################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/share/space.go:62]: Not needed

File: cmd/vclusterctl/cmd/platform/sleep/space.go [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+#######################################################
+################ devspace sleep space #################
+#######################################################
+Sleep puts a space to sleep
+Example:
+devspace sleep space myspace
+devspace sleep space myspace --project myproject
+#######################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/sleep/space.go:62]:
```suggestion
```

File: cmd/vclusterctl/cmd/platform/wakeup/space.go [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+#######################################################
+################ devspace wakeup space ################
+#######################################################
+wakeup resumes a sleeping space
+Example:
+devspace wakeup space myspace
+devspace wakeup space myspace --project myproject
+#######################################################
+	`
+	}
```
Comment (ThomasK33) [inline cmd/vclusterctl/cmd/platform/wakeup/space.go:51]:
```suggestion
```

**Review by ThomasK33 — CHANGES_REQUESTED** [review]

**Review by ThomasK33 — APPROVED** [review]

**Pull request merged**